src/features/defi/providers/osmosis/components/OsmosisManager/Deposit/components/Deposit.tsx

```diff
@@ -0,0 +1,402 @@
+      underlyingAsset0,
+      underlyingAsset1,
+      calculateAllocations,
+      getDepositFeeEstimate,
+      onNext,
+      toast,
+      translate,
+    ],
+  )
+
+  useEffect(() => {
+    // Fetch pool data
+    ;(async () => {
+      if (!(state && state.opportunity)) return undefined
+
+      const { assetReference: poolAssetReference } = fromAssetId(state.opportunity.assetId)
+      const id = getPoolIdFromAssetReference(poolAssetReference)
+      if (!id) return undefined
+
+      if (!poolData) setPoolData(await getPool(id))
+    })()
+  }, [poolData, state])
+
+  const handleCancel = browserHistory.goBack
+
+  const handleBack = useCallback(() => {
+    history.push({
+      pathname: `/defi/earn`,
+      search: qs.stringify({
+        ...query,
+        modal: DefiAction.Overview,
+      }),
+    })
+  }, [history, query])
+
+  if (
+    !(
+      asset &&
+      state &&
+      contextDispatch &&
+      osmosisOpportunity &&
+      underlyingAsset0 &&
+      underlyingAsset1 &&
+      underlyingAsset0MarketData &&
+      underlyingAsset1MarketData
+    )
+  ) {
+    return null
+  }
+
+  const validateCryptoAmount = (value: string, isForAsset1: boolean) => {
+    const crypto = bnOrZero(
+      isForAsset1 ? underlyingAsset0Balance : underlyingAsset1Balance,
+    ).dividedBy(bn(10).pow((isForAsset1 ? underlyingAsset0 : underlyingAsset1).precision ?? '0'))
+    const _value = bnOrZero(value)
+    const hasValidBalance = crypto.gt(0) && _value.gt(0) && crypto.gte(value)
+    if (_value.isEqualTo(0)) return ''
+    return hasValidBalance || 'common.insufficientFunds'
+  }
+
+  const validateFiatAmount = (value: string, isForAsset1: boolean) => {
+    const crypto = bnOrZero(isForAsset1 ? underlyingAsset0Balance : underlyingAsset1Balance).div(
+      bn(10).pow((isForAsset1 ? underlyingAsset0 : underlyingAsset1).precision),
+    )
+    const fiat = crypto.times(
+      (isForAsset1 ? underlyingAsset0MarketData : underlyingAsset1MarketData).price,
+    )
+    const _value = bnOrZero(value)
+    const hasValidBalance = fiat.gt(0) && _value.gt(0) && fiat.gte(value)
+    if (_value.isEqualTo(0)) return ''
+    return hasValidBalance || 'common.insufficientFunds'
+  }
+
+  const underlyingAsset0CryptoAmountAvailable = bnOrZero(underlyingAsset0Balance).div(
+    bn(10).pow(underlyingAsset0.precision),
+  )
+
+  const underlyingAsset1CryptoAmountAvailable = bnOrZero(underlyingAsset1Balance).div(
+    bn(10).pow(underlyingAsset1.precision),
+  )
+
+  return (
+    <PairDepositWithAllocation
+      accountId={accountId}
+      asset1={underlyingAsset0}
+      asset2={underlyingAsset1}
+      icons={osmosisOpportunity?.icons}
+      destAsset={asset}
+      apy={osmosisOpportunity?.apy?.toString() ?? ''}
+      calculateAllocations={calculateAllocations}
+      cryptoAmountAvailable1Precision={underlyingAsset0CryptoAmountAvailable.toPrecision()}
+      cryptoAmountAvailable2Precision={underlyingAsset1CryptoAmountAvailable.toPrecision()}
```

Comment:
Ditto this can be avoided by selecting the opportunity in `<PairDepositWithAllocation />`